Widgets/ColorPicker.cpp

```diff
@@ -0,0 +1,37 @@
+#include "ColorPicker.h"
+
+#include <QColorDialog>
+#include <QHBoxLayout>
+#include <QPainter>
+
+ColorPicker::ColorPicker(QWidget *parent) : QWidget(parent) {
+  QHBoxLayout *layout = new QHBoxLayout(this);
+  layout->setMargin(0);
+  button = new QToolButton(this);
+  layout->addWidget(button);
+  this->setLayout(layout);
+
+  connect(button, &QToolButton::clicked, [=]() {
+    QColor color = QColorDialog::getColor(this->color_, this, tr("Choose a color"), QColorDialog::ShowAlphaChannel);
+    if (!color.isValid()) return;
+    this->setColor(color);
+  });
+  connect(this, &ColorPicker::colorChanged, this, &ColorPicker::updateButtonIcon);
+  this->updateButtonIcon();
+}
+
+QColor ColorPicker::color() const { return this->color_; }

```

Comment:
No need for this `this->`. Also in three places below.

Reply:
Fair enough: 1575b9f1ae7f877cf2e97a9cda43da857f8eec84